### PR TITLE
:fire::tada:  Sticker Block Redesign #638

### LINKED ISSUE #638: :lipstick: Sticker Block Redesign: block redesign
## User stories

As a user, I want the "Sticker Block" component in the app to be redesigned to enhance its usability and visual appeal.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Technical Description
The current design is: 
image

It should be changed to:
image

The files can be found at `libs/features/convs-mgr/stories/blocks/library/sticker-message-block/`. 

The designs are [here](https://www.figma.com/file/ZGHzSstXOYPMFU1EiqjEHD/Untitled?type=design&node-id=1%3A4&mode=design&t=Hf7DxaUUMQ3BWy57-1)

### Acceptance Criteria

1. The block should look exactly like the designs.
2. Use of `rem`, `em` as opposed to `px`.
3. All changes should be isolated to the following  `libs/features/convs-mgr/stories/blocks/library/sticker-message-block/`.

### PR DESCRIPTION
**Pull request by kiplangat01**
# Description

This PR fixes issue #638 Sticker Block Redesign

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes #638 

# Screenshot 
Screenshot from 2023-11-14 10-41-18

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/sticker-block.jsplumb.ts [hunk 1]
```diff
@@ -21,8 +21,17 @@ export function _StickerBlockDecoratePlumb(block: StickerMessageBlock, comp: Com
...
+  });
+
+  jsPlumb.addEndpoint(comp.location.nativeElement, {
+    // Whether the anchor is target (Other Block -> This Block)
+    target: true,
+    cssClass:"block_endpoint",
+    endpoint: "Dot",
+    anchor: [1, 0.7, 1, 0],
     maxConnections: -1
   });
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/sticker-block.jsplumb.ts:36]: remove the extra endpoint and then return the default option.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Sync the latest changes then request review

File: libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts [hunk 2]
```diff
@@ -4,9 +4,10 @@ import { FormGroup } from '@angular/forms';
 import { take } from 'rxjs';
 import { SubSink } from 'subsink';
 import { BrowserJsPlumbInstance } from '@jsplumb/browser-ui';
-
+import { iconsAndTitles } from '../../../../../main/src/lib/model/icons-and-titles';
 import { FileStorageService } from '@app/state/file';
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts:8]: icon is selected from the parent block component

File: libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts [hunk 3]
```diff
@@ -26,20 +27,26 @@ export class StickerBlockComponent implements OnInit, OnDestroy {
...
     const fileSize = this.stickerMessageForm.get('fileSize')?.value;
 
     if (fileSize) {
       this._checkSizeLimit(fileSize);
     }
   }
-
+  getBlockIconAndTitle(type: number) {
+    return iconsAndTitles[type];
+  }
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts:49]: icon is selected from parent

File: libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts [hunk 4]
```diff
@@ -26,20 +27,26 @@ export class StickerBlockComponent implements OnInit, OnDestroy {
...
+  svgIcon = ''
   private _sBs = new SubSink();
 
   constructor(private _stickerUploadService: FileStorageService) {}
 
   ngOnInit(): void {
     this.stickerInputId = `stckr-${this.id}`;
 
+    this.svgIcon = this.getBlockIconAndTitle(this.block.type).svgIcon
+
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts:40]: same here. should be accessed from the parent block component

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

File: libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts (hunk 2 above)
Comment (kiplangat01) [reply libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts:8]: this is to handle the icon in the middle of block where you click to upload file

File: libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts (hunk 4 above)
Comment (kiplangat01) [reply libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts:40]: this is to handle the icon in the middle of block where you click to upload file

Comment (LemmyMwaura) [reply libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts]: ah, okay that makes sense. leave it as is.

File: libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts (hunk 2 above)
Comment (LemmyMwaura) [reply libs/features/convs-mgr/stories/blocks/library/sticker-message-block/src/lib/components/sticker-block/sticker-block.component.ts]: Okay, nevermind. Leave it as is.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

**Comment by LemmyMwaura** [comment]
sticker

The design is off and the icon is not centered

**Review by LemmyMwaura — APPROVED** [review]

**Pull request merged**